src/Processors/Transforms/buildPushingToViewsChain.cpp

```diff
@@ -186,6 +186,151 @@ class FinalizingViewsTransform final : public IProcessor
     std::exception_ptr any_exception;
 };
 
+Block getHeader(ContextPtr context, ASTPtr query)
+{
+    if (context->getSettingsRef().allow_experimental_analyzer)
+        return InterpreterSelectQueryAnalyzer::getSampleBlock(query, context);
+    else
+        return InterpreterSelectQuery(query, context, SelectQueryOptions().analyze()).getSampleBlock();
+}
+
+
+std::optional<Chain> generateViewChain(
+    const StorageID & view_id,
+    ContextPtr context,
+    ContextPtr select_context,
+    ContextPtr insert_context,
+    ThreadGroupPtr running_group,
+    Chain & result_chain,
+    ViewsDataPtr views_data,
+    ThreadStatusesHolderPtr thread_status_holder,
+    bool async_insert,
+    const Block & storage_header)
+{
+    auto view = DatabaseCatalog::instance().tryGetTable(view_id, context);
+    if (view == nullptr)
+    {
+        LOG_WARNING(
+            &Poco::Logger::get("PushingToViews"), "Trying to access table {} but it doesn't exist", view_id.getFullTableName());
+        return std::nullopt;
+    }
+
+    auto view_metadata_snapshot = view->getInMemoryMetadataPtr();
+
```

Comment:
Please make a comment about how you're going to use `local_select_context` and `local_insert_context`.